src/app/archive/archive_lib/processor.ml

```diff
@@ -107,6 +107,44 @@ module Snarked_ledger_hash = struct
           hash
 end
 
+module Epoch_data = struct
+  type t = {seed: string; ledger_hash_id: int}
+
+  let typ =
+    let encode t = Ok (t.seed, t.ledger_hash_id) in
+    let decode (seed, ledger_hash_id) = Ok {seed; ledger_hash_id} in
+    let rep = Caqti_type.(tup2 string int) in
+    Caqti_type.custom ~encode ~decode rep
+
+  let add_if_doesn't_exist (module Conn : CONNECTION)
+      (t : Coda_base.Epoch_data.Value.t) =
+    let open Deferred.Result.Let_syntax in
+    let Coda_base.Epoch_ledger.Poly.{hash; _} =
+      Coda_base.Epoch_data.Poly.ledger t
+    in
+    let%bind ledger_hash_id =
```

Comment:
should this snarked ledger hash already exist, in all cases? (I'm not sure)

if so, we can just do a query instead of adding it